pkg/build/api/v1/types.go

```diff
@@ -620,6 +620,9 @@ type GenericWebHookEvent struct {
 
 	// Git is the git information if the Type is BuildSourceGit
 	Git *GitInfo `json:"git,omitempty"`
+
+	// Env contains additional environment variables you want to pass into a builder container
+	Env []kapi.EnvVar `json:"env,omitempty"`
```

Comment:
On Tue, Apr 26, 2016 at 1:36 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/api/v1/types.go
> <https://github.com/openshift/origin/pull/8477#discussion_r61129731>:
>
> > @@ -620,6 +620,9 @@ type GenericWebHookEvent struct {
> >
> >  	// Git is the git information if the Type is BuildSourceGit
> >  	Git *GitInfo `json:"git,omitempty"`
> > +
> > +	// Env contains additional environment variables you want to pass into a builder container
> > +	Env []kapi.EnvVar `json:"env,omitempty"`
>
> @bparees <https://github.com/bparees> - just to be sure, are you
> referring to AllowEnv being a boolean toggle we would add to
> https://github.com/openshift/origin/blob/master/pkg/build/api/types.go#L573-L576
> ?
>
> that was my thinking, yes, although that struct is shared between generic
> and git webhook definitions, so it would mean the AllowEnv field would have
> no meaning when users were defining github webhooks. not ideal.
>
Is having 2 WebHookTrigger types, where we potentially update the spots
where the single type is leveraged to deal with 2 types, the lesser of the
evils ?

> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8477/files/fde0d585e7272ad1f545600043154b5afd9a9cd0#r61129731>
>